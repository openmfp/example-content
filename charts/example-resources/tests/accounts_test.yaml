suite: accounts
templates:
  - accounts/*.yaml
tests:
  - it: matches the snapshot
    set:
      accounts.enabled: true
    asserts:
      - matchSnapshot: {}
