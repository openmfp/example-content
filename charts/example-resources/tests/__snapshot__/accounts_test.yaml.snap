matches the snapshot:
  1: |
    apiVersion: core.openmfp.io/v1alpha1
    kind: Account
    metadata:
      name: demo
      namespace: demo-root
    spec:
      displayName: Demo Account
      type: account
  2: |
    apiVersion: v1
    kind: Namespace
    metadata:
      name: openmfp-root
  3: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: configmap-manager
    rules:
      - apiGroups:
          - ""
        resources:
          - configmaps
        verbs:
          - get
          - list
          - watch
          - create
          - update
          - patch
          - delete
  4: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: configmap-manager-binding
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: configmap-manager
    subjects:
      - kind: ServiceAccount
        name: account-operator
        namespace: openmfp-system
  5: |
    apiVersion: core.openmfp.io/v1alpha1
    kind: Account
    metadata:
      name: demo-root
      namespace: openmfp-root
    spec:
      displayName: OpenMFP Root Account
      extensions:
        - apiVersion: v1
          kind: ConfigMap
          specGoTemplate: {}
      namespace: demo-root
      type: folder
